### ISSUE TITLE
Variation in features extracted by capa while uisng vivisect backend and capa IDA explorer

### ISSUE THREAD

**Issue by r0ny123** [fireeye/capa#518]
I've noticed that for a particular sample (`70da6872b6b2da9ddc94d14b02302917`), the no/type of features getting extracted by capa while using `vivisect` and `capa explorer`, are not same. there's some significant mismatch between the features extracted by above mentioned methods.

![image](https://user-images.githubusercontent.com/49360849/113913789-5b0b0880-97fa-11eb-8f46-eaac3228ed87.png)

**left**:_IDA capa explorer results_ **right**:_extracted by capa using vivisect_

For example, `receive data on socket (3 matches)` and `send data on socket (3 matches)` (on left) is not present in right screenshot.

Is this something expected? I don't know atm, if this occurring for other files too.

**Comment by williballenthin** [comment]
Capa relies on these analysis backends (IDA, vivisect, SMDA, ...) to identify functions, basic blocks, instructions, features, etc. If the backend engines disagree, then there's not too much that capa can do. If you can highlight a few samples with differing analysis, then we can triage why the backends differ and maybe supply patches to those engines to enhance their analysis. Otherwise, unfortunately, its not really in scope to fix this within capa (however, if there's an issue in how we extract from the backend engine, thats definitely a bug! please do share samples with differing analysis for this reason).

**Comment by r0ny123** [comment]
Yeah, I already mentioned the hash `70da6872b6b2da9ddc94d14b02302917`

**Comment by r0ny123** [comment]
Is there any way to know how many features get extracted while running capa on IDA backend?

**Comment by mr-tz** [comment]
Also be aware that the standalone tool does some filtering to display less hits. In the example above `send data` is a "parent" rule for `send data on socket` so the latter does not get displayed, for example.

To get a feature count you can use `scripts/show-featues.py` in IDA or use the save feature in the plugin and inspect the resulting JSON.

**Comment by r0ny123** [comment]
> Also be aware that the standalone tool does some filtering to display less hits. In the example above send data is a "parent" rule for send data on socket so the latter does not get displayed, for example.

I didn't notice that. Later, I ran `capa -v <file>`and  `receive data on socket (3 matches)`, `send data on socket (3 matches)` were there with their parent one. But do we need to display the parent one while the children are already displayed both in `IDA capa exlorer` and terminal, while running `capa` from CLI with `-v`.

> To get a feature count you can use scripts/show-featues.py in IDA or use the save feature in the plugin and inspect the resulting JSON.

Thanks!

**Comment by mr-tz** [comment]
I think this has been answered. Please reopen if you still have issues.
